src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+const char D_801DFAF4[] = "h";
+const char D_801DFAF8[] = "i";
+const char D_801DFAFC[] = "f";
+const char D_801DFB00[] = "fa";
+const char D_801DFB04[] = "fb";
+const char D_801DFB08[] = "fc";
+const char D_801DFB0C[] = "fd";
+const char D_801DFB10[] = "fe";
+const char D_801DFB14[] = "ff";
+const char D_801DFB18[] = "fg";
+const char D_801DFB1C[] = "fh";
+const char D_801DFB20[] = "fi";
+const char D_801DFB24[] = "fj";
+const char D_801DFB28[] = "fk";
+
+void Play_ClearTransition(PlayState* this) {
+    if (this->transitionCtx.fbdemoType != -1) {
+        Transition_Destroy(&this->transitionCtx);
+    }
+    this->transitionCtx.transitionType = -1;
+}
+
+Gfx* Play_SetFog(PlayState* this, Gfx* gfx) {
+    s32 fogFar = this->lightCtx.fogFar * (5.0f / 64.0f);
+
+    return Gfx_SetFogWithSync(gfx, this->lightCtx.fogColor.r, this->lightCtx.fogColor.g, this->lightCtx.fogColor.b, 0,
+                              this->lightCtx.fogNear, (fogFar <= 1000) ? 1000 : fogFar);
+}
+
+void Play_Destroy(GameState* thisx) {
+    PlayState* this = (PlayState*)thisx;
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+
+    if (sBombersNotebookOpen) {
+        MsgEvent_SendNullTask();
+        func_80178750();
+        gfxCtx->curFrameBuffer = SysCfb_GetFbPtr(gfxCtx->framebufferIdx % 2);
+        gfxCtx->zbuffer = SysCfb_GetZBuffer();
+        gfxCtx->viMode = D_801FBB88;
+        gfxCtx->viConfigFeatures = gViConfigFeatures;
+        gfxCtx->xScale = gViConfigXScale;
+        gfxCtx->yScale = gViConfigYScale;
+        gfxCtx->updateViMode = true;
+        sBombersNotebookOpen = false;
+    }
+
+    Hireso_Destroy(&sHireso);
+    this->state.gfxCtx->callback = NULL;
+    this->state.gfxCtx->callbackParam = 0;
+    Play_DestroyMotionBlur();
+
+    if (R_PAUSE_MENU_MODE != 0) {
+        PreRender_ApplyFiltersSlowlyDestroy(&this->pauseBgPreRender);
+        R_PAUSE_MENU_MODE = 0;
+    }
+
+    R_PICTOGRAPH_PHOTO_STATE = PICTOGRAPH_PHOTO_STATE_OFF;
+    PreRender_Destroy(&this->pauseBgPreRender);
+    this->unk_18E58 = NULL;
+    this->pictoPhotoI8 = NULL;
+    this->unk_18E60 = NULL;
+    this->unk_18E64 = NULL;
+    this->unk_18E68 = NULL;
+    Effect_DestroyAll(this);
+    EffectSS_Clear(this);
+    CollisionCheck_DestroyContext(this, &this->colChkCtx);
+
+    if (gTrnsnUnkState == 3) {
+        TransitionUnk_Destroy(&sTrnsnUnk);
+        gTrnsnUnkState = 0;
+    }
+
+    if ((this->transitionMode == TRANS_MODE_INSTANCE_RUNNING) || D_801D0D54) {
+        this->transitionCtx.destroy(&this->transitionCtx.instanceData);
+        Play_ClearTransition(this);
+        this->transitionMode = TRANS_MODE_OFF;
+    }
+
+    ShrinkWindow_Destroy();
+    TransitionFade_Destroy(&this->unk_18E48);
+    VisMono_Destroy(&sVisMono);
+    func_80140EA0(D_801F6D4C);
+    D_801F6D4C = NULL;
+
+    if (gSaveContext.save.weekEventReg[0x5C] & 0x80) {
+        Actor_CleanupContext(&this->actorCtx, this);
+    }
+    gSaveContext.save.weekEventReg[0x5C] &= (u8)~0x80;
```

Comment:
This is a very curious flag, since the only place it is set seems to be on `Actor_InitContext`, which I guess it makes sense considering what it does, but it doesn't make sense it is a weekEventReg